src/main/java/org/elasticsearch/cluster/ClusterService.java

```diff
@@ -120,4 +120,10 @@
      */
     int numberOfPendingTasks();
 
+    /**
+     * Returns the wait time for the first task in the pending task queue
+     *
+     * @returns A zero time value if the queue is empty, otherwise the time value of the task waiting in the queue
+     */
+    TimeValue getPendingTaskTimeInQueue();
```

Comment:
I think this should be maxPendingTaksTimeInQueue. Watch out that this isn't necessarily based on the first task in the queue since this is a priority queue. Left some notes about the implementation as well.